vernac/himsg.ml

```diff
@@ -149,8 +149,22 @@ let explicit_flags =
     [print_implicits; print_coercions; print_no_symbol]; (** Then more! *)
     [print_universes; print_implicits; print_coercions; print_no_symbol] (** and more! *) ]
 
+let with_diffs pm pn =
+  if Proof_diffs.color_enabled () then begin
+    try
+      let tokenize_string = Proof_diffs.tokenize_string in
+      Pp_diff.diff_pp ~tokenize_string pm pn
+    with Pp_diff.Diff_Failure msg ->
+      Feedback.msg_warning Pp.(str ("Diff failure: " ^ msg) ++ cut()
+          ++ str "Showing message without diff highlighting" );
```

Comment:
I did put warning like that one in the STM before, and I did regret it.
The problem is that this warning makes sense only to a developer.
Also, a warning should come with a way to silence it, and in this case a user cannot (he can disable the whole feature, but I guess it does not count).

Reply:
I think it's unlikely users will ever see this message.  If it's ever seen, I would want to get a bug report and then make a fix.  The message would appear if the strings cannot be lexed.  https://github.com/coq/coq/pull/8734 tweaks the lexer behavior to make that even less likely.

I could check whether some environment variable is defined, and if so, suppress the warning message.  But not documented so that users would (hopefully) report the problem.

What do you think?

Reply:
If you want users to report the bug, you need to tell them to do so like we do with anomalies.